src/core/timer.c

```diff
@@ -845,6 +845,41 @@ static void timer_timezone_change(Unit *u) {
         }
 }
 
+static int timer_clean(Unit *u, ExecCleanMask mask) {
+        Timer *t = TIMER(u);
+        int r;
+
+        assert(t);
+        assert(mask != 0);
+
+        if (t->state != TIMER_DEAD)
+                return -EBUSY;
+
+        if (!IN_SET(mask, EXEC_CLEAN_STATE))
+                return -EUNATCH;
+
+        r = timer_setup_persistent(t);
+        if (r < 0)
+                return r;
+
+        if (!t->stamp_path)
+                return -EUNATCH;
+
+        if (unlink(t->stamp_path) && errno != ENOENT)
```

Comment:
IIUC, this code will run on PID1 main loop. Timer stamps live in /var/lib, and thus this could block for arbitrarily large periods of time?

Reply:
the timestamp file is already written to from PID 1 too... While we don't try to go overboard with /var accesses from PID 1 there are a number of them in there. This wasn't different in sysvinit and other init systems btw, for example to maintain the utmp/wtmp stuff (which we maintain too).